test/jdk/com/sun/net/httpserver/bugs/TcpNoDelayNotRequired.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2005, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 6968351
+ * @summary  tcp no delay not required for small payloads
+ * @library /test/lib
+ * @run main/othervm/timeout=5 -Dsun.net.httpserver.nodelay=false  TcpNoDelayNotRequired
```

Comment:
I think we should remove the `timeout=5` here. In the past we have seen that such timeouts have contributed to intermittent failures in the CI. jtreg itself has a (sufficiently large) timeout and if the test doesn't complete by then, then jtreg errors that test as timed out. Relying on jtreg timeout handling will avoid guessing the right timeout value here in the test definition.

Reply:
The timeout is the test. These changes fix the performance by an order of 100x. The timeout is the only way I know to test this.